### PR TITLE
Support multiple .apk files produced from gradle build

### PR DESCRIPTION
**Pull request by Fatme**
This PR adds support for multiple .apk files produced from gradle build in following cases:

1. When productFlavors are used
 Add this to `app/App_Resources/app.gradle` file
```
flavorDimensions "arch", "mode"

  productFlavors {
    demo {
      dimension "mode"
    }
    full {
      dimension "mode"
    }
    x86 {
      dimension "arch"
    }
    arm {
      dimension "arch"
    }
    arm64 {
      dimension "arch"
    }
  }
```

2. When splits option is used
Add this to `app/App_Resources/app.gradle` file
```
splits {
    abi {
      enable true //enables the ABIs split mechanism
      reset() //reset the list of ABIs to be included to an empty string
      include 'arm64-v8a', 'armeabi-v7a', 'x86'
      universalApk true
    }
  }
```

### CODE REVIEW

**Comment by Fatme** [comment]
@rosen-vladimirov @Mitko-Kerezov

File: lib/services/platform-service.ts [hunk 1]
```diff
@@ -746,27 +746,50 @@ export class PlatformService extends EventEmitter implements IPlatformService {
...
+		return null;
+	}
+
+	private createApplicationPackages(packages: string[]): IApplicationPackage[] {
+		return packages.map(filepath => this.createApplicationPackage(filepath));
 	}
 
-	private getLatestApplicationPackage(buildOutputPath: string, validPackageNames: string[]): IApplicationPackage {
-		let packages = this.getApplicationPackages(buildOutputPath, validPackageNames);
+	private createApplicationPackage(filepath: string): IApplicationPackage {
```
Comment (Mitko-Kerezov) [inline lib/services/platform-service.ts]: This method's argument could be named `packageName` in order to enable a shorter syntax down below:
```ts
return {
    packageName,
    time: this.$fs.getFsStats(packageName).mtime
};
```

File: test/platform-commands.ts [hunk 2]
```diff
@@ -35,10 +35,8 @@ class PlatformData implements IPlatformData {
 	};
 	emulatorServices: Mobile.IEmulatorPlatformServices = null;
 	projectRoot = "";
-	getDeviceBuildOutputPath = (buildTypeOption: IRelease) => {
-		return "";
-	}
-	getValidPackageNames = (buildOptions: { isForDevice?: boolean, isReleaseBuild?: boolean }) => [""];
+	deviceBuildOutputPath = "";
+	getValidBuildOutputData = (buildOptions: { isForDevice?: boolean, isReleaseBuild?: boolean }) => ({ packageNames: [""] });
```
Comment (Mitko-Kerezov) [inline test/platform-commands.ts]: An interface can be used here as well

File: test/stubs.ts [hunk 3]
```diff
@@ -417,10 +415,8 @@ export class PlatformsDataStub extends EventEmitter implements IPlatformsData {
 			projectRoot: "",
 			normalizedPlatformName: "",
 			appDestinationDirectoryPath: "",
-			getDeviceBuildOutputPath: (buildTypeOption: IRelease) => {
-				return "";
-			},
-			getValidPackageNames: (buildOptions: { isForDevice?: boolean, isReleaseBuild?: boolean }) => [],
+			deviceBuildOutputPath: "",
+			getValidBuildOutputData: (buildOptions: { isForDevice?: boolean, isReleaseBuild?: boolean }) => ({ packageNames: []}),
```
Comment (Mitko-Kerezov) [inline test/stubs.ts]: An interface may be reused here as well

File: test/stubs.ts [hunk 4]
```diff
@@ -304,10 +304,8 @@ export class PlatformProjectServiceStub extends EventEmitter implements IPlatfor
 			platformProjectService: this,
 			emulatorServices: undefined,
 			projectRoot: "",
-			getDeviceBuildOutputPath: (buildTypeOption: IRelease) => {
-				return "";
-			},
-			getValidPackageNames: (buildOptions: { isForDevice?: boolean, isReleaseBuild?: boolean }) => [],
+			deviceBuildOutputPath: "",
+			getValidBuildOutputData: (buildOptions: { isForDevice?: boolean, isReleaseBuild?: boolean }) => ({ packageNames: [] }),
```
Comment (Mitko-Kerezov) [inline test/stubs.ts]: An interface may be reused here as well

**Comment by dtopuzov** [comment]
run ci

**Comment by Fatme** [comment]
run ci

**Comment by Fatme** [comment]
run ci

**Pull request merged**